src/slarray.jl

```diff
@@ -1,16 +1,16 @@
-struct SLArray{S,N,Syms,T} <: StaticArray{S,T,N}
-  __x::SArray{S,T,N}
+struct SLArray{S,N,L,Syms,T} <: StaticArray{S,T,N}
+  __x::SArray{S,T,N,L}
   #SLArray{Syms}(__x::StaticArray{S,T,N}) where {S,N,Syms,T} = new{S,N,Syms,T}(__x)
-  SLArray{S,N,Syms,T}(__x::SArray) where {S,N,Syms,T} = new{S,N,Syms,T}(T.(__x))
-  SLArray{S,N,Syms}(x::Tuple) where {S,N,Syms} = new{S,N,Syms,eltype(x)}(SArray{S,eltype(x),N}(x))
-  SLArray{S,N,Syms,T}(x::Tuple) where {S,N,Syms,T} = new{S,N,Syms,T}(SArray{S,T,N}(T.(x)))
+  Base.@pure SLArray{S,N,Syms,T}(__x::SArray) where {S,N,Syms,T} = new{S,N,length(__x),Syms,T}(convert.(T,__x))
```

Comment:
Should this be `ndims`?
```julia
julia> @SArray ones(3,2)
3×2 SArray{Tuple{3,2},Float64,2,6}:
 1.0  1.0
 1.0  1.0
 1.0  1.0

julia> ndims(ones(3,2))
2
```

Reply:
I don't think `@pure` should be used of functions with multiple methods. Can't we delegate most of the constructor logic to the `StaticArrays` constructors instead of redoing it here?

Reply:
we should delegate it to StaticArrays as much as possible.